stablehlo/dialect/StablehloOps.td

```diff
@@ -1118,6 +1118,17 @@ def StableHLO_IfOp: StableHLO_Op<"if", [
                         SizedRegion<1>:$false_branch);
 
   let results = (outs Variadic<HLO_TensorOrToken>);
+
+  let extraClassDeclaration = [{
+    // Method from InferTypeOpInterface interface.
+    static bool isCompatibleReturnTypes(TypeRange l, TypeRange r) {
```

Comment:
Is this different from `isCompatibleForTypeRange` from above? Maybe that's what Eugene's feedback from above meant.

Reply:
Exactly, and this method is copied again and again among different ops in ODS. That's why the pending https://github.com/openxla/stablehlo/pull/145 is trying to address this issue by moving this function to `Base.h/cc` and adding to top level op then inherit by all ops. 
Like Eugene said: https://github.com/openxla/stablehlo/pull/145 may need some time to refine and land, so it does not hurt if I copy this method to `Base.h/cc` in advance in the PR right now.